mesonbuild/modules/i18n.py

```diff
@@ -202,6 +202,9 @@ def _get_source_files(self, sources: T.Iterable[SourcesType]) -> T.Set[mesonlib.
                 source_files.update(source.get_sources())
             elif isinstance(source, build.BothLibraries):
                 source_files.update(source.get('shared').get_sources())
+            elif isinstance(source, (build.CustomTarget, build.CustomTargetIndex)):
+                FeatureNew.single_use('i18n.xgettext with custom_target is broken until 1.10', '1.10.0', self.interpreter.subproject, location=self.interpreter.current_node)
```

Comment:
This should really go in the `xgettext` interpreter method.

Additionally, `CustomTargetIndex` is a new feature in 1.10.0, so that needs a FeatureNew as well.